cmd/image/sync/sync.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 IBM Corp
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sync
+
+import (
+	"errors"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+)
+
+// constant for creating cos instance
+const (
+	ServiceType = "cloud-object-storage"
+)
+
+var Cmd = &cobra.Command{
+	Use:   "sync",
+	Short: "Sync images between IBM COS buckets",
+	Long: `Sync images between IBM COS buckets
+pvsadm image sync --help for information
+
+# Set the API key or feed the --api-key commandline argument
+export IBMCLOUD_API_KEY=<IBM_CLOUD_API_KEY>
+
+Examples:
+
+# using spec yaml file
+pvsadm image sync --spec-file spec.yaml
+
+Sample spec.yaml file:
+---
+- source:
+    bucket: bucket-fqsdgh
+    cos: cos-test-buphij
+    object: ".txt"
+    storageClass: smart
+    region: jp-tok
+  target:
+  - bucket: bucket-ielfqq
+    storageClass: smart
+    region: jp-tok
+  - bucket: bucket-kkoyrg
+    storageClass: smart
+    region: jp-tok
+- source:
+    bucket: bucket-embocx
+    cos: cos-test-icrbul
+    object: ""
+    storageClass: smart
+    region: us-east
+  target:
+  - bucket: bucket-xgskog
+    storageClass: standard
+    region: jp-tok
+  - bucket: bucket-nomoer
+    storageClass: cold
+    region: jp-tok
+
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var s3Cli *client.S3Client
+		var selectedObjects []string
+		type copyWorkload struct {
+			s3Cli     client.S3Client
+			srcBucket string
+			tgtBucket string
+			srcObject string
+		}
+
+		const NoOfCopyWorkers = 20
```

Comment:
Its better to declare the constants  at the beginning of the file with a line of description

Reply:
Moved to constants section at the beginning of the file.